api/search.rb

```diff
@@ -1,104 +1,138 @@
-get "#{APIPREFIX}/search/threads" do
-  local_params = params # Necessary for params to be available inside blocks
-  group_ids = get_group_ids_from_params(local_params)
-  context = local_params["context"] ? local_params["context"] : "course"
-  search_text = local_params["text"]
-  if !search_text
-    {}.to_json
-  else
-    # Because threads and comments are currently separate unrelated documents in
-    # Elasticsearch, we must first query for all matching documents, then
-    # extract the set of thread ids, and then sort the threads by the specified
-    # criteria and paginate. For performance reasons, we currently limit the
-    # number of documents considered (ordered by update recency), which means
-    # that matching threads can be missed if the search terms are very common.
+def get_thread_ids(context, group_ids, local_params, search_text)
+  filters = []
+  filters.push({term: {commentable_id: local_params['commentable_id']}}) if local_params['commentable_id']
+  filters.push({terms: {commentable_id: local_params['commentable_ids'].split(',')}}) if local_params['commentable_ids']
+  filters.push({term: {course_id: local_params['course_id']}}) if local_params['course_id']
 
-    get_matching_thread_ids = lambda do |search_text|
-      self.class.trace_execution_scoped(["Custom/get_search_threads/es_search"]) do
-        search = Tire.search Content::ES_INDEX_NAME do
-          query do
-            match [:title, :body], search_text, :operator => "AND"
-            filtered do
-              filter :term, :commentable_id => local_params["commentable_id"] if local_params["commentable_id"]
-              filter :terms, :commentable_id => local_params["commentable_ids"].split(",") if local_params["commentable_ids"]
-              filter :term, :course_id => local_params["course_id"] if local_params["course_id"]
-              filter :or, [
-                {:not => {:exists => {:field => :context}}},
-                {:term => {:context => context}}
-              ]
+  filters.push({or: [
+      {not: {exists: {field: :context}}},
+      {term: {context: context}}
+  ]})
+
+  unless group_ids.empty?
+    filters.push(
+        {
+            or: [
+                {:not => {:exists => {:field => :group_id}}},
+                {:terms => {:group_id => group_ids}}
+            ]
+        }
+    )
+  end
 
-              if not group_ids.empty?
-                if group_ids.length > 1
```

Comment:
Thanks @clintonb.  I wasn't sure if you might remember something around this.

It's possible that me and @dianakhuang can work this out, especially since she knows about the group ids.  I'll let you know if we need more help.